### PR TITLE
nftables support

### PR DESCRIPTION
**Pull request by carlhoerberg**
Direct integration with nftables, so that a process doesn't have to spawned for each accepted connection, which is resource intensive on overloaded servers

### CODE REVIEW

File: src/nftables.cr [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  def finalize
+    LibNftables.nft_ctx_free(@nft)
+  end
+
+  def run_cmd(cmd : String) : Nil
+    buf = Bytes.new(cmd.bytesize + 1) # null terminated string
+    buf.copy_from(cmd.to_slice)
+    LibNftables.nft_run_cmd_from_buffer(@nft, buf).zero? ||
+      raise Error.new("nftables command could not be executed")
```
Comment (jage) [inline src/nftables.cr]: Include command in error message?

**Comment by jage** [comment]
Nice! Haven't tested but looks good!

File: src/nftables.cr (hunk 1 above)
Comment (carlhoerberg) [reply src/nftables.cr]: yes, thanks

**Review by jage — APPROVED** [review]
Maybe add something about nftables libs in READEM, like `libnftables-dev on Ubuntu to build server`. Haven't done any real testing, but code looks reasonable. :)

**Pull request merged**